### PR TITLE
Ripul handoo/relationship/pod distribution budget

### PR DESCRIPTION
**Pull request by RipulHandoo**
**Notes for Reviewers**

This PR fixes #
1. Relationship between Pod Distribution Budget and other K8s resources.
2. fixes path for event component.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by RipulHandoo** [comment]
Working Proof of this PR.

https://github.com/meshery/meshery/assets/107461226/9fd4215a-e7ce-457e-829b-3d1b9fd1aa51

**Comment by RipulHandoo** [comment]
Working Proof for Relationship between Event and other k8s resources. 

https://github.com/meshery/meshery/assets/107461226/591869ed-a7bf-479b-b017-07390890038d

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -371,10 +371,10 @@
...
               "patchStrategy": "replace",
               "mutatedRef": [
-                ["settings", "metadata", "name"],
-                ["settings", "metadata", "namespace"],
-                ["settings", "metadata", "kind"],
-                ["settings", "metadata", "apiVersion"]
+                ["settings", "regarding", "name"],
+                ["settings", "regarding", "namespace"],
+                ["settings", "regarding", "kind"],
+                ["settings", "regarding", "apiVersion"]
```
Comment (leecalcote) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:377]: eh? Really?

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 2]
```diff
@@ -560,6 +560,73 @@
...
+              "description": "StatefulSets are used for stateful applications that require stable, unique network identifiers and persistent storage. PDBs can be used with StatefulSets to control pod disruptions and ensure data integrity during maintenance or scaling."
+            }
+          }, 
+          {
+            "kind": "DaemonSet",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatorRef": [
+                ["settings", "spec", "selector", "matchLabels"]
```
Comment (MUzairS15) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:606]: This should be `labels`.
As PDB will acquire/controls pods which have the same labels as specified inside PDB's `selector.matchLabels`

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json (hunk 1 above)
Comment (leecalcote) [reply server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:377]: @RipulHandoo ?

Comment (RipulHandoo) [reply server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:377]: Yes @leecalcote, regarding contains the **_object_** this Event is about. 
There is one more field `related` but that is optional (means it is used when regarding object triggers the creation and deletion of related object).

Example: 
```
apiVersion: v1
kind: Event
metadata:
  name: deployment-event
  namespace: default
  labels:
    type: deployment-related
    component: deployment-controller
data:
  message: "Deployment 'example-deployment' scaled successfully"
  deploymentName: example-deployment
  eventType: Normal
regarding:
  apiVersion: apps/v1
  kind: Deployment
  name: example-deployment
  namespace: default
  ```

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json (hunk 2 above)
Comment (RipulHandoo) [reply server/meshmodel/kubernetes/relationships/hierarchical_inventory.json]: Done.

**Comment by leecalcote** [comment]
Merge conflicts, @RipulHandoo

**Comment by leecalcote** [comment]
Let's start breaking out these relationships into different files at some point.

**Review by MUzairS15 — APPROVED** [review]
LGTM!

**Pull request merged**